File: app/schemas/subscription.py
from pydantic import BaseModel, HttpUrl
from typing import Optional, List
from datetime import datetime

class SubscriptionBase(BaseModel):
    target_url: HttpUrl
    event_types: Optional[List[str]] = None

class SubscriptionCreate(SubscriptionBase):
    secret_key: str

class SubscriptionUpdate(SubscriptionBase):
    pass

class Subscription(SubscriptionBase):
    id: int
    is_active: bool
    created_at: datetime
    updated_at: Optional[datetime]

    class Config:
        from_attributes = True

File: webhook-frontend/src/types/subscription.ts
export interface Subscription {
    id: number;
    target_url: string;
    event_types: string[];
    is_active: boolean;
    created_at: string;
    updated_at: string | null;
}

export interface SubscriptionCreate {
    target_url: string;
    secret_key: string;
    event_types: string[];
}

